python/cugraph/graph/c_graph.pyx

```diff
@@ -280,56 +269,52 @@ class Graph:
         # still in use inside this class. If copy is set to True, deep-copy the
         # objects.
         if copy is False:
-            self.adj_list_offset_col = offset_col;
-            self.adj_list_index_col = index_col;
-            self.adj_list_value_col = value_col;
+            self.adj_list_offset_col = offset_col
+            self.adj_list_index_col = index_col
+            self.adj_list_value_col = value_col
         else:
-            self.adj_list_offset_col = offset_col.copy();
-            self.adj_list_index_col = index_col.copy();
-            self_adj_list_value_col = value_col.copy();
+            self.adj_list_offset_col = offset_col.copy()
+            self.adj_list_index_col = index_col.copy()
+            self_adj_list_value_col = value_col.copy()
 
         cdef uintptr_t graph = self.graph_ptr
-        cdef uintptr_t offsets = create_column(self.adj_list_offset_col)
-        cdef uintptr_t indices = create_column(self.adj_list_index_col)
-        cdef uintptr_t value
+        cdef gdf_column c_offset_col = get_gdf_column_view(self.adj_list_offset_col)
+        cdef gdf_column c_index_col = get_gdf_column_view(self.adj_list_index_col)
+        cdef gdf_column c_value_col
+        cdef gdf_column * c_value_col_ptr
         if value_col is None:
-            value = 0
+            c_value_col_ptr = NULL
         else:
-            value = create_column(self.adj_list_value_col)
+            c_value_col = get_gdf_column_view(self.adj_list_value_col)
```

Comment:
Is there a scoping issue here? You are creating an object on the stack inside of a code block and assigning a pointer to it which is used outside of the code block.

Reply:
```python
283         cdef gdf_column c_value_col
```

c_value_col is defined in line 238, and it's not used after

```python
291         err = gdf_adj_list_view(<gdf_graph*> graph,
292                                 &c_offset_col,
293                                 &c_index_col,
294                                 c_value_col_ptr)
```

Please note that the actual column array data are stored in self.adj_list_value_col (a python object), and c_value_col is a temporary C++ wrapper used to pass the column data in python object to a C++ function expecting C++ data type.

Reply:
It seems to me that `c_value_col` is still in the scope when used in `gdf_adj_list_view`.
However, there is also a small consistency issue here as `c_offset_col` and `c_index_col` are accessed from the stack objects while a pointer to the stack object is created for the values.

Reply:
The inconsistency is because value is optional (it can be NULL). I can either use value_col_ptr or replicate the gdf_edge_list_view call for the two different cases (if value_col is None and else). I picked to use value_col_ptr.